app/Traits/SubscriptionsTrait.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace App\Traits;
+
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Relations\HasOne;
+use Illuminate\Support\Facades\DB;
+use Laravel\Cashier\Subscription;
+use Wave\PaddleSubscription;
+use Wave\Plan;
+
+trait SubscriptionsTrait
+{
+    protected string $paymentVendor;
+
+    public function __construct()
```

Comment:
constructor in trait is a dangerous and tricky. Please consider to use approach with cached getPaymentVendor() method instead

Reply:
Updated